src/agent.cpp

```diff
@@ -62,6 +62,31 @@ namespace Rep
 
     std::string Agent::escape(const std::string& query)
     {
-        return Url::Url(query).defrag().escape().fullpath();
+        // Special case: treat multiple leading slashes as one.
+        std::string query_;
+        if (query.size() > 1 && query[0] == '/')
```

Comment:
I would add a comment saying this assumes that the query has already been stripped of leading and trailing whitespace. This precondition is being maintained by the code in `Robots::getpair` which is invoked prior to calling `Agent::disallowed` or `Agent::allowed`.

Honestly, I would probably document that `Agent::escape` is really an internal function to `Agent` even though it is currently a public static function on that class.